tests/pds/resiliency_test.go

```diff
@@ -56,29 +56,111 @@ var _ = Describe("{KillDeploymentControllerDuringDeployment}", func() {
 		StartTorpedoTest("KillDeploymentControllerDuringDeployment", "Kill Deployment Controller Pod when a DS Deployment is happening", pdsLabels, 0)
 	})
 
+	It("Deploy Dataservices", func() {
+		Step("Deploy Data Services", func() {
+			for _, ds := range params.DataServiceToTest {
+				Step("Start deployment, Kill Deployment Controller Pod while deployment is ongoing and validate data service", func() {
+					isDeploymentsDeleted = false
+					// Global Resiliency TC marker
+					pdslib.MarkResiliencyTC(true, false)
+					// Type of failure that this TC needs to cover
+					failuretype := pdslib.TypeOfFailure{
+						Type: KillDeploymentControllerPod,
+						Method: func() error {
+							return pdslib.KillPodsInNamespace(params.InfraToTest.PDSNamespace, pdslib.PdsDeploymentControllerManagerPod)
+						},
+					}
+					pdslib.DefineFailureType(failuretype)
+					// Deploy and Validate this Data service after injecting the type of failure we want to catch
+					deployment, _, _, err = TriggerDeployDataService(ds, params.InfraToTest.Namespace, tenantID, projectID)
+					log.FailOnError(err, "Error while deploying data services")
+					err = pdslib.InduceFailureAfterWaitingForCondition(deployment, namespace, params.ResiliencyTest.CheckTillReplica)
+					log.FailOnError(err, fmt.Sprintf("Error happened while executing Kill Deployment Controller test for data service %v", *deployment.ClusterResourceName))
+				})
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		pdslib.CloseResiliencyChannel()
+		if !isDeploymentsDeleted {
```

Comment:
It will delete the latest deployments only, we need to add a step to delete created deployments after line 78